src/lumigo_tracer/spans_container.py

```diff
@@ -286,12 +287,16 @@ def create_span(cls, event=None, context=None, force=False) -> "SpansContainer":
         """
         if cls._span and not force:
             return cls._span
+        # copy the event to ensure that we will not change it
+        event = copy.deepcopy(event)
```

Comment:
it's on purpise - 
This one is here to make sure that we will not change the parameters for the customers.
The one on the parser is because this specific parser changed the value, and we want to continue using the original one...

Reply:
Look like the Parser should be pure.. NVM

Reply:
agree...
This makes it pure :)